pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -257,6 +259,26 @@ func convertHTTPRoute(r k8s.HTTPRouteRule, ctx configContext,
 		}
 	}
 
+	if r.Timeouts != nil {
+		if r.Timeouts.Request != nil {
+			request, _ := time.ParseDuration(string(*r.Timeouts.Request))
+			if request != 0 {
```

Comment:
does GW api allow explicitly setting 0?

If not, should we default to no timeout?

Reply:
The spec requires explicit setting to 0 for no timeout. The default is implementation defined, so I think we should just do what Istio/envoy does by default.